src/org/traccar/protocol/Pt502ProtocolEncoder.java

```diff
@@ -15,11 +15,27 @@
  */
 package org.traccar.protocol;
 
+import java.util.TimeZone;
+
 import org.traccar.StringProtocolEncoder;
 import org.traccar.helper.Log;
 import org.traccar.model.Command;
 
-public class Pt502ProtocolEncoder extends StringProtocolEncoder {
+public class Pt502ProtocolEncoder extends StringProtocolEncoder implements StringProtocolEncoder.ValueFormatter {
+
+    @Override
+    public String formatValue(String key, Object value) {
+        if (key.equals(Command.KEY_TIMEZONE)) {
+            return String.valueOf(TimeZone.getTimeZone(value.toString()).getRawOffset() / 3600000);
+        }
+
+        return null;
+    }
+
+    @Override
+    protected String formatCommand(Command command, String format, String... keys) {
+        return super.formatCommand(command, format, this, keys);
```

Comment:
Why `super`?

Reply:
Actually, I've copied from here https://github.com/tananaev/traccar/blob/master/src/org/traccar/protocol/WatchProtocolEncoder.java#L45
Should I remove both?

Reply:
I think originally it was same signature in watch encoder, so super was required. Now it looks like it's not required. Please use it as well.